File: app/src/main/java/org/wikipedia/analytics/FindInPageFunnel.java
package org.wikipedia.analytics;

import android.support.annotation.NonNull;

import org.json.JSONObject;
import org.wikipedia.WikipediaApp;
import org.wikipedia.dataclient.WikiSite;

public class FindInPageFunnel extends TimedFunnel {
    private static final String SCHEMA_NAME = "MobileWikiAppFindInPage";
    private static final int REV_ID = 14586774;

    private final int pageId;
    private int pageHeight;
    private int numFindNext;
    private int numFindPrev;
    private String findText;

    public FindInPageFunnel(WikipediaApp app, WikiSite wiki, int pageId) {
        super(app, SCHEMA_NAME, REV_ID, Funnel.SAMPLE_LOG_ALL, wiki);
        this.pageId = pageId;
    }

    @Override protected void preprocessSessionToken(@NonNull JSONObject eventData) { }

    public void setPageHeight(int height) {
        this.pageHeight = height;
    }

    public void addFindNext() {
        numFindNext++;
    }

    public void addFindPrev() {
        numFindPrev++;
    }

    public void setFindText(String text) {
        findText = text;
    }

    public void logDone() {
        log(
                "pageID", pageId,
                "numFindNext", numFindNext,
                "numFindPrev", numFindPrev,
                "findText", findText,
                "pageHeight", pageHeight
        );
    }
}

File: app/src/main/java/org/wikipedia/analytics/FindInPageFunnel.kt
package org.wikipedia.analytics

import org.json.JSONObject
import org.wikipedia.WikipediaApp
import org.wikipedia.dataclient.WikiSite

class FindInPageFunnel(app: WikipediaApp, wiki: WikiSite?, private val pageId: Int) :
        TimedFunnel(app, SCHEMA_NAME, REV_ID, SAMPLE_LOG_ALL, wiki) {

    private var numFindNext = 0
    private var numFindPrev = 0
    var pageHeight = 0
    var findText: String? = null

    override fun preprocessSessionToken(eventData: JSONObject) {}

    fun addFindNext() {
        numFindNext++
    }

    fun addFindPrev() {
        numFindPrev++
    }

    fun logDone() {
        log(
                "pageID", pageId,
                "numFindNext", numFindNext,
                "numFindPrev", numFindPrev,
                "findText", findText,
                "pageHeight", pageHeight
        )
    }

    companion object {
        private const val SCHEMA_NAME = "MobileWikiAppFindInPage"
        private const val REV_ID = 19690671
    }
}
